src/glyph/serialize.rs

```diff
@@ -84,6 +97,71 @@ impl Glyph {
     }
 }
 
+fn libs_to_object_libs(glyph: &Glyph) -> Plist {
+    let mut object_libs = Plist::default();
+
```

Comment:
we could write less code by declaring a little closure here to handle the repeated adding of an id/lib pair:

```rust
let mut add_lib |id: Option<&Identifer>, lib: &Dictionary| {
    let id = id.map(|id| id.as_str().to_string());
    debug_assert!(id.is_some());
    object_libs.insert(id.unwrap(), plist::Value::Dictionary(lib.clone()));
};
```

And then elsewhere we just call this, like:

```rust
for anchor in anchor {
    if let Some(lib) = anchor.lib() {
        add_lib(anchor.identifier(), lib);
    }
}
```


On `debug_assert!`:

`debug_assert!` doesn't make a lot of sense here, since we're going to unwrap anyway. `debug_assert!` would make more sense if we were going to just skip these items in a release build or something like that? What do we actually want to do in this case?

Reply:
I just wanted to have a non-crpytic error message if this invariant somehow fails but I'm going to remove it :smile_cat: need to give error handling some deeper thought anyway.